packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts

```diff
@@ -104,6 +105,9 @@ export const useNotifiSubscribe: () => Readonly<{
       setPhoneNumber(phoneNumber ?? '');
 
       const telegramTarget = targetGroup?.telegramTargets[0];
+      if (telegramTarget !== null && emailTarget?.isConfirmed === false) {
```

Comment:
Ah, I think you are looking for `if (telegramTarget !== null && telegramTarget?.isConfirmed === false)`.

Actually, a `telegramConfirmationUrl` should not return back if the telegram Id has been confirmed. Instead, in the `NotifiTelegramInput`, you can expose the `telegramConfirmationUrl` and check if it is null or not.

Reply:
double checked with the code in notifi hub, yeah, you are right, we can use telegramConfirmationUrl to check if the telegram has been confirmed or not.